File: CS-Examples/05_Outlook Storage File/GetSubFoldersInfo.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Spire.Email;
using Spire.Email.Outlook;
using System.IO;

namespace GetSubFoldersInfo
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void btnRun_Click(object sender, EventArgs e)
        {
            //Load PST file
            OutlookFile olf = new OutlookFile(@"..\..\..\..\..\..\Data\Sample.pst");
            //Get the folders collection
            OutlookFolderCollection folderCollection = olf.RootOutlookFolder.GetSubFolders();
            StringBuilder sb = new StringBuilder();
            //Display folder name and number of message
            foreach(OutlookFolder folderinfo in folderCollection)
            {
                sb.AppendLine("Folder:" + folderinfo.Name);
                sb.AppendLine("Total items:" + folderinfo.ItemCount);
                sb.AppendLine("Total unread item:" + folderinfo.UnreadItemCount);
                sb.AppendLine("------------------Next Folder--------------------");
            }
            File.WriteAllText("GetSubFoldersInfo.txt", sb.ToString());
            MessageBox.Show("Completed");
        }
    }
}

File: VB-Examples/05_Outlook Storage File/GetSubFoldersInfo.vb
Imports System.ComponentModel
Imports System.Text
Imports Spire.Email
Imports Spire.Email.Outlook
Imports System.IO

Namespace GetSubFoldersInfo
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
			'Load PST file
            Dim olf As New OutlookFile("..\..\..\..\..\..\Data\Sample.msg")
			'Get the folders collection
			Dim folderCollection As OutlookFolderCollection = olf.RootOutlookFolder.GetSubFolders()
            Dim sb As New StringBuilder()
            'Display folder name and number of message
			For Each folderinfo As OutlookFolder In folderCollection
                sb.AppendLine("Folder:" & folderinfo.Name)
                sb.AppendLine("Total items:" & folderinfo.ItemCount)
                sb.AppendLine("Total unread item:" & folderinfo.UnreadItemCount)
                sb.AppendLine("------------------Next Folder--------------------")
            Next folderinfo
            File.WriteAllText("GetSubFoldersInfo.txt", sb.ToString)
            MessageBox.Show("Completed")
		End Sub
	End Class
End Namespace